app/src/main/java/com/duckduckgo/app/sitepermissions/SitePermissionsActivity.kt

```diff
@@ -0,0 +1,126 @@
+import com.duckduckgo.site.permissions.store.sitepermissions.SitePermissionsEntity
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@InjectWith(ActivityScope::class)
+class SitePermissionsActivity : DuckDuckGoActivity() {
+
+    @Inject
+    lateinit var faviconManager: FaviconManager
+
+    private val viewModel: SitePermissionsViewModel by bindViewModel()
+    private val binding: ActivityLocationPermissionsBinding by viewBinding()
+    private val adapter: SitePermissionsAdapter by lazy { SitePermissionsAdapter(viewModel, this, faviconManager) }
+
+    private val toolbar
+        get() = binding.includeToolbar.toolbar
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(binding.root)
+        setupToolbar(toolbar)
+        setupRecyclerView()
+        observeViewModel()
+        viewModel.allowedSites()
+    }
+
+    private fun observeViewModel() {
+        lifecycleScope.launch {
+            viewModel.viewState
+                .flowWithLifecycle(lifecycle, STARTED)
+                .collectLatest { state ->
+                    val sitePermissionsWebsites =
+                        state.locationPermissionsAllowed.map { it.domain }.union(state.sitesPermissionsAllowed.map { it.domain }).toList()
+                    updateList(sitePermissionsWebsites, state.askLocationEnabled, state.askCameraEnabled, state.askMicEnabled)
+                }
+        }
+        lifecycleScope.launch {
+            viewModel.commands
+                .flowWithLifecycle(lifecycle, STARTED)
+                .collectLatest { processCommand(it) }
+        }
+    }
+
+    private fun processCommand(command: Command) {
+        when (command) {
+            is ShowRemovedAllConfirmationSnackbar -> showRemovedAllSnackbar(command.removedSitePermissions, command.removedLocationPermissions)
+            is LaunchWebsiteAllowed -> launchWebsiteAllowed(command.domain)
+        }
+    }
+
+    private fun showRemovedAllSnackbar(
+        removedSitePermissions: List<SitePermissionsEntity>,
+        removedLocationPermissions: List<LocationPermissionEntity>
+    ) {
+        val message = HtmlCompat.fromHtml(getString(R.string.sitePermissionsRemoveAllWebsitesSnackbarText), HtmlCompat.FROM_HTML_MODE_LEGACY)
+        Snackbar.make(
+            binding.root,
+            message,
+            Snackbar.LENGTH_LONG
+        ).setAction(R.string.fireproofWebsiteSnackbarAction) {
+            viewModel.onSnackBarUndoRemoveAllWebsites(removedSitePermissions, removedLocationPermissions)
+        }.show()
+    }
+
+    private fun updateList(
+        sitesAllowed: List<String>,
+        askLocationEnabled: Boolean,
+        askCameraEnabled: Boolean,
+        askMicEnabled: Boolean
+    ) {
+        adapter.updateItems(sitesAllowed, askLocationEnabled, askCameraEnabled, askMicEnabled)
```

Comment:
If we decide only requested permissions should show in the screen, we will need to change this logic here.